src/dreyfus_index_updater.erl

```diff
@@ -89,3 +101,43 @@ load_docs(FDI, _, {I, IndexPid, Db, Proc, Total, LastCommitTime}=Acc) ->
         false ->
             {ok, setelement(1, Acc, I+1)}
     end.
+
+purge_index(Db, IndexPid, Index) ->
+    {ok, IdxPurgeSeq} = clouseau_rpc:get_purge_seq(IndexPid),
+    Proc = get_os_process(Index#index.def_lang),
+    true = proc_prompt(Proc, [<<"add_fun">>, Index#index.def]),
+    FoldFun = fun(PurgeSeq, {Id, _Revs}, Acc) ->
+        case couch_db:get_full_doc_info(Db, Id) of
+            not_found ->
+                ok = clouseau_rpc:delete(IndexPid, Id),
+                Acc0 = Acc;
+            FDI ->
+                DI = couch_doc:to_doc_info(FDI),
+                #doc_info{id=Id, revs=[#rev_info{rev=Rev, deleted=Del}|_]} = DI,
+                case lists:member({Id, Rev}, Acc) of
+                    true ->
+                        Acc0 = Acc;
+                    false ->
+                        case Del of
+                            true ->
+                                ok = clouseau_rpc:delete(IndexPid, Id);
+                            false ->
+                                {ok, Doc} = couch_db:open_doc(Db, DI, []),
+                                Json = couch_doc:to_json_obj(Doc, []),
+                                [Fields|_] = proc_prompt(Proc, [<<"index_doc">>, Json]),
+                                Fields1 = [list_to_tuple(Field) || Field <- Fields],
+                                case Fields1 of
+                                    [] ->
+                                        ok = clouseau_rpc:delete(IndexPid, Id);
+                                    _  ->
+                                        ok = clouseau_rpc:update(IndexPid, Id, Fields1)
+                                end
+                        end,
+                        lists:keydelete(Id, 1, Acc),
+                        Acc0 = lists:append([{Id, Rev}], Acc)
```

Comment:
`[{Id, Rev} | Acc]` is more idiomatic (and faster). If order matters, then reverse the list at the end.